### PR TITLE
Improve punishment handling for shared IPs

### PR DESCRIPTION
**Pull request by bumbadadabum**

### CODE REVIEW

File: users.js [hunk 1]
```diff
@@ -1086,16 +1086,18 @@ class User {
 		this.inRooms.clear();
 	}
 	getAlts(includeTrusted, forPunishment) {
-		return this.getAltUsers(includeTrusted, forPunishment).map(user => user.getLastName());
+		return this.getAltUsers(includeTrusted, forPunishment).filter(user => user.named).map(user => user.getLastName());
```
Comment (Zarel) [inline users.js]: `getLastName` can extract previous names from users not currently named, though.

Comment (bumbadadabum) [inline users.js]: Oh, what happens in the case of a user that has only been on a guest account though?

Comment (Zarel) [inline users.js]: Then it falls back to `Guest [...]`, but this should be rare enough not to really be a problem.

Comment (bumbadadabum) [inline users.js]: ah k. editing that now

File: chat-commands.js [hunk 2]
```diff
@@ -1286,7 +1286,7 @@ exports.commands = {
 		this.addModCommand(`${name} was banned from ${room.title} by ${user.name}.${reason}`, ` (${targetUser.latestIp})`);
 
 		if (!room.isPrivate && room.chatRoomData) {
-			let alts = targetUser.getAlts();
+			let alts = targetUser.getAlts(false, true);
```
Comment (Zarel) [inline chat-commands.js]: The problem here is that it'll include self... Maybe `.slice(1)`?

Comment (bumbadadabum) [inline chat-commands.js]: Maybe, yeah

**Comment by bumbadadabum** [comment]
Made the changes you requested @Zarel

**Comment by bumbadadabum** [comment]
Ok I also did something else. This PR is kinda broader than the original intention now, so I changed the title accordingly.

File: punishments.js [hunk 3]
```diff
@@ -409,9 +410,10 @@ Punishments.punish = function (user, punishment, recursionKeys) {
 	}
 
 	let keys = recursionKeys || new Set();
+	let affected;
```
Comment (xfix) [inline punishments.js]: Not sure why scope of `affected` variable was changed.

Comment (bumbadadabum) [inline punishments.js]: Because affected is used in two different if-statements now.

Comment (Zarel) [inline punishments.js]: Yeah, I guess that's not clear unless you expand the diff.

**Pull request merged**